### PR TITLE
Refactor logging to log transaction ids instead of RPT ids

### PR DESCRIPTION
**Pull request by giovanniberti**
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java [hunk 1]
```diff
@@ -324,7 +324,7 @@ void shouldThrowAlreadyProcessedOn401ForCreditCardWithXpay() throws JsonProcessi
         StepVerifier.create(client.requestXPayAuthorization(authorizationData))
                 .expectErrorMatches(
                         error -> error instanceof AlreadyProcessedException &&
-                                ((AlreadyProcessedException) error).getRptId()
+                                ((AlreadyProcessedException) error).getTransactionId()
                                         .equals(transaction.getNoticeCodes().get(0).rptId())
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java:328]: Here should equality be tested against transactions id and not 1st notices RPT ID?

File: src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java [hunk 2]
```diff
@@ -400,7 +400,7 @@ void shouldThrowAlreadyProcessedOn401ForPostePay() throws JsonProcessingExceptio
         StepVerifier.create(client.requestPostepayAuthorization(authorizationData))
                 .expectErrorMatches(
                         error -> error instanceof AlreadyProcessedException &&
-                                ((AlreadyProcessedException) error).getRptId()
+                                ((AlreadyProcessedException) error).getTransactionId()
                                         .equals(transaction.getNoticeCodes().get(0).rptId())
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java:404]: Same as above

**Review by pietro-tota — APPROVED** [review]
Well done! Just some comment. Also there is a compilation error with the `TransactionAddUserReceiptHandler`, did you miss the genaral refactoring for this class?

File: src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java (hunk 2 above)
Comment (giovanniberti) [reply src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java]: Whoops! Missed that 😅  Fixed in c84d279

File: src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java (hunk 1 above)
Comment (giovanniberti) [reply src/test/java/it/pagopa/transactions/client/PaymentGatewayClientTest.java]: (you're right, see below, fixed in c84d279)

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**